File: library/src/main/java/com/thefuntasty/taste/keyboard/TKeyboard.java
package com.thefuntasty.taste.keyboard;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public class TKeyboard {
	private TKeyboard() {
	}

	/**
	 * Hides the soft keyboard
	 */
	public static void hideSoftKeyboard(Activity activity) {
		View focusedView = activity.getCurrentFocus();
		if (focusedView != null) {
			InputMethodManager inputMethodManager = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
			inputMethodManager.hideSoftInputFromWindow(focusedView.getWindowToken(), 0);
		}
	}

	/**
	 * Hides the soft keyboard
	 */
	public static void hideSoftKeyboard(Context context, View focusedView) {
		if (focusedView != null) {
			InputMethodManager inputMethodManager = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
			inputMethodManager.hideSoftInputFromWindow(focusedView.getWindowToken(), 0);
		}
	}

	/**
	 * Shows the soft keyboard
	 */
	public static void showSoftKeyboard(View view) {
		InputMethodManager inputMethodManager = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
		view.requestFocus();
		inputMethodManager.showSoftInput(view, 0);
	}
}

File: core/src/main/java/com/thefuntasty/taste/keyboard/TKeyboard.kt
package com.thefuntasty.taste.keyboard

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager

class TKeyboard {

	fun hide(activity: Activity) {
		val focusedView = activity.currentFocus
		hide(focusedView)
	}

	fun hide(focusedView: View?) {
		if (focusedView != null) {
			val inputMethodManager = focusedView.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
			inputMethodManager.hideSoftInputFromWindow(focusedView.windowToken, 0)
		}
	}

	fun show(view: View) {
		val inputMethodManager = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
		view.requestFocus()
		inputMethodManager.showSoftInput(view, 0)
	}
}
